src/test/java/com/internship/backend/model/NewDateTest.java

```diff
@@ -0,0 +1,203 @@
+        assertTrue(newDate3.isBefore(newDate1));
+        assertTrue(newDate2.isBefore(newDate1));
+        assertFalse(newDate1.isBefore(newDate2));
+    }
+
+    @Test
+    void testSameDateComparison(){
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+
+        assertFalse(newDate1.isAfter(newDate1));
+        assertFalse(newDate1.isBefore(newDate1));
+    }
+
+    @Test
+    void getMinute() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1.getMinute(), 30);
+        assertEquals(newDate2.getMinute(), 30);
+        assertEquals(newDate3.getMinute(), 0);
+        assertNotEquals(newDate1.getMinute(), 12);
+    }
+
+    @Test
+    void getHour() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1.getHour(), 12);
+        assertEquals(newDate2.getHour(), 10);
+        assertEquals(newDate3.getHour(), 14);
+        assertNotEquals(newDate1.getHour(), 15);
+    }
+
+    @Test
+    void getDay() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1.getDay(), 5);
+        assertEquals(newDate2.getDay(), 5);
+        assertEquals(newDate3.getDay(), 20);
+        assertNotEquals(newDate1.getDay(), 15);
+    }
+
+    @Test
+    void getMonth() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1.getMonth(), 3);
+        assertEquals(newDate2.getMonth(), 3);
+        assertEquals(newDate3.getMonth(), 2);
+        assertNotEquals(newDate1.getMonth(), 15);
+    }
+
+    @Test
+    void getYear() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2028);
+        NewDate newDate3 = new NewDate(0,14,20,2,2025);
+
+        assertEquals(newDate1.getYear(), 2024);
+        assertEquals(newDate2.getYear(), 2028);
+        assertEquals(newDate3.getYear(), 2025);
+        assertNotEquals(newDate1.getYear(), 2003);
+    }
+
+    @Test
+    void setMinute() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2028);
+        NewDate newDate3 = new NewDate(0,14,20,2,2025);
+
+        newDate1.setMinute(10);
+        newDate2.setMinute(15);
+        newDate3.setMinute(7);
+
+        assertEquals(newDate1.getMinute(), 10);
+        assertEquals(newDate2.getMinute(), 15);
+        assertEquals(newDate3.getMinute(), 7);
+        assertNotEquals(newDate1.getMinute(), 12);
+    }
+
+    @Test
+    void setHour() {
```

Comment:
In general methods like setter/getter are not required to be tested. We usually write tests for the methods that have logic on it.